### PR TITLE
Fixed the way completed task count is computed and some refactoring

### PR DESCRIPTION
**Pull request by liyinan926**
Signed-off-by: Yinan Li <liyinan926@gmail.com>

### CODE REVIEW

**Comment by liyinan926** [comment]
@zliu41 can you review?

File: gobblin-runtime/src/main/java/gobblin/runtime/TaskState.java [hunk 1]
```diff
@@ -169,6 +169,16 @@ public void setTaskDuration(long duration) {
   }
 
   /**
+   * Return whether the task has completed running or not.
+   *
+   * @return {@code true} if the task has completed or {@code false} otherwise
+   */
+  public boolean isCompleted() {
+    WorkingState state = getWorkingState();
+    return state == WorkingState.SUCCESSFUL || state == WorkingState.COMMITTED || state == WorkingState.FAILED;
```
Comment (zliu41) [inline gobblin-runtime/src/main/java/gobblin/runtime/TaskState.java]: Should CANCELLED be included?

Comment (liyinan926) [inline gobblin-runtime/src/main/java/gobblin/runtime/TaskState.java]: CANCELLED tasks are semantically not completed so that's why it's not included here.

Comment (zliu41) [inline gobblin-runtime/src/main/java/gobblin/runtime/TaskState.java]: In `LocalJobManager.onTaskCompletion()`, it won't commit the job unless `jobState.getCompletedTasks() == jobState.getTaskCount()`. If a CANCELLED task doesn't count for completed tasks but does count for total task count, will it prevent the job from being committed?

Comment (liyinan926) [inline gobblin-runtime/src/main/java/gobblin/runtime/TaskState.java]: `LocalJobManager` has been long deprecated and shouldn't be used.

**Comment by zliu41** [comment]
LGTM

**Pull request merged**